src/main/java/uk/gov/hmcts/reform/cpo/repository/CasePaymentOrdersRepository.java

```diff
@@ -1,16 +1,23 @@
 package uk.gov.hmcts.reform.cpo.repository;
 
+import org.springframework.transaction.annotation.Transactional;
+import uk.gov.hmcts.reform.cpo.errorhandling.CasePaymentIdentifierException;
 import org.springframework.data.domain.Page;
 import org.springframework.data.domain.Pageable;
-import org.springframework.data.jpa.repository.JpaRepository;
-import org.springframework.stereotype.Repository;
 import uk.gov.hmcts.reform.cpo.data.CasePaymentOrderEntity;
 
 import java.util.List;
 import java.util.UUID;
 
-@Repository
-public interface CasePaymentOrdersRepository extends JpaRepository<CasePaymentOrderEntity, UUID> {
+@Transactional(rollbackFor = CasePaymentIdentifierException.class)
+public interface CasePaymentOrdersRepository {
```

Comment:
minor, probably not worth changing now:
wonder whether this unified repository, which combines both the CPO and CPOAudit repositories is not a complication. When you add a method to one of the repos, you need to add it also in here. 
Can't the service just be injected with both CPO and CPOAudit repositories? And the additional logic required in some of these methods be included in the Service class

Reply:
Good point - will leave for now as you said and see how code looks when other endpoints and associated services and repositories have been added